src/SFA.DAS.EmployerCommitmentsV2/Extensions/IIDentityExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Security.Principal;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerCommitmentsV2.Web.Extensions
+{
+    public static class IIdentityExtensions
+    {
+        public static string Upn(this IIdentity identity)
+        {
+            switch (identity)
+            {
+                case ClaimsIdentity claimsIdentity:
+                    return claimsIdentity.Claims.FirstOrDefault(claim => claim.Type == EmployeClaims.Upn)?.Value;
```

Comment:
I think this claim type is only applicable for the Provider not the employer (different IDAMS)

Reply:
I copied this across pre-emptively for CV-190. I'll check this as part of that story (I'm working on it now) and fix it then if the claims are different.

Reply:
The name space is wrong for this class, and the filename is ever-so-slightly-wrong, as is the `EmployeClaims` class.

The claims themselves (looking at v1 code) seem to be:
`  public static class DasClaimTypes
  {
    private const string BaseUrl = "http://das/employer/identity/claims/";
    public const string Id = "http://das/employer/identity/claims/id";
    public const string Email = "http://das/employer/identity/claims/email_address";
    public const string GivenName = "http://das/employer/identity/claims/given_name";
    public const string FamilyName = "http://das/employer/identity/claims/family_name";
    public const string DisplayName = "http://das/employer/identity/claims/display_name";
    public const string RequiresVerification = "http://das/employer/identity/claims/requires_verification";
  }`
plus, "sub" which appears to be the User Id.